### ISSUE TITLE
`type` and `Type` are narrowed inconsistently when using `isinstance`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6252]
This is based on a mypy bug report: https://github.com/python/mypy/issues/16349.

Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgFDkDGANgIYDOdUAwgFylQdQB0P5pAJsWBRgAVxQUAjAAoYkloRIBKNp0zCkdVHRg0JxWZIA0zFezWcQxAG7EaVAPrwEBuUr6DhYiQCZZPhWdiMzUkDS0UHT0KVx8TJhCLDitbeycSf3cgA)

```python
from typing import Type

class C:
    ...

def func1(t1: Type):
    if isinstance(t1, C):
        reveal_type(t1)

def func2(t2: type):
    if isinstance(t2, C):
        reveal_type(t2)
```

**Comment by erictraut** [comment]
The solution to this depends in part on the conclusion to [this discussion in the Python Typing forum](https://discuss.python.org/t/inconsistencies-between-type-and-type/37404).

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.338, which I just published. It will also be included in a future release of pylance.
